usr/lib/ep11_stdll/new_host.c

```diff
@@ -1493,24 +1525,108 @@ CK_RV SC_Login(STDLL_TokData_t *tokdata, ST_SESSION_HANDLE *sSession,
         }
 
         rc = load_masterkey_so(tokdata);
-        if (rc != CKR_OK)
+        if (rc != CKR_OK) {
             TRACE_DEVEL("Failed to load SO's masterkey.\n");
+        }
+        break;
+    case CKU_CONTEXT_SPECIFIC:
```

Comment:
Looks like this code block is identical in common/new_host.c and ep11stdll/new_host.c. Is there a way to move this to a common subroutine?

Reply:
EP11's new_host.c is a copy of the common new_host.c (similar for ICSF's new_host.c). So they are 95% equal.... I don't like this too much, but extracting everything common into subroutines is a major task. I have it on my long-term TODO list, but I am not sure if this will ever be done :-) 
Currently the goal is to keep these files as equal as possible, where only the token specific things differ.